homeassistant/components/nad/media_player.py

```diff
@@ -200,12 +201,16 @@ def update(self):
         self._volume = self.calc_volume(self._nad_receiver.main_volume("?"))
         self._source = self._source_dict.get(self._nad_receiver.main_source("?"))
 
-    def calc_volume(self, decibel):
+    def calc_volume(self, decibel: int) -> Optional[float]:
```

Comment:
`decibel` can't be `None` according to the type hint. It's probably better to catch this before calling `calc_volume`.

Reply:
Good point, but I added the type hint. The decibel comes directly from the upstream nad_receiver package and that does report None under some circumstances. I'd even argue that is correct, since some receivers have no idea about the volume (min has a stepper motor which can move the volume up and down but doesn't know how to report the current volume sadly).

Reply:
Yes. So we should check for `None` before calling `self.calc_volume` instead of inside it.